### ISSUE TITLE
Medtrum Nano

### ISSUE THREAD

**Issue by misterizzz** [nightscout/Trio#261]
because of certain benefits of the Medtrum Nano, compared with Omnipod, an integration in Trio would be great.

**Comment by JeremyStorring** [comment]
Thanks for the feature request. I actually only found out about Medtrum Nano a few hours ago (totally unrelated to work here). 

It looks like it's already implemented / will be implemented with AAPS, so it shouldn't be completely impossible to add here.

Adding link for future work.
https://github.com/nightscout/AndroidAPS/pull/2623

**Comment by LiroyvH** [comment]
I believe @jbr7rr is the maintainer for the kotlin driver and has been seeking some assistance in getting it ported to swift and integrated.

**Comment by Sjoerd-Bo3** [comment]
Yeah would be nice indeed. But would also be a lot of work ofc. We need someone that is up for that.

**Comment by jbr7rr** [comment]
> Yeah would be nice indeed. But would also be a lot of work ofc. We need someone that is up for that.

See my reply to this issue here:
https://github.com/LoopKit/Loop/issues/2111#issuecomment-2099248381

**Comment by WesSec** [comment]
I'd be willing to sponsor development for this. The only reason I'm still carrying around an Android phone

**Comment by Sjoerd-Bo3** [comment]
> I'd be willing to sponsor development for this. The only reason I'm still carrying around an Android phone

Thats very kind of you, but that was not the problem. It’s a big task to take on and someone has to do it. 
We are looking into it though. But it will take time, so please have patience.
If you still want to sponsor, please sponsor The Nightscout Foundation. We can always apply for funds there aswell, might we need it.

**Comment by anthonyplaisir** [comment]
Hello any more informations to integrate medtrum to IAPS?

**Comment by Sjoerd-Bo3** [comment]
> Hello any more informations to integrate medtrum to IAPS?

Nope, it will come in due time. But expect months to end of the year. We are still in the starting stages.

**Comment by Anti-style** [comment]
Hi, i just stumbled on this thread. How is the work going? I will start using a medtrum nano soon and have an iPhone. So I would be interested in using the medtrum with IAPS as well. Thanks for the hard work anyway!

**Comment by Sjoerd-Bo3** [comment]
> Hi, i just stumbled on this thread. How is the work going? I will start using a medtrum nano soon and have an iPhone. So I would be interested in using the medtrum with IAPS as well. Thanks for the hard work anyway!

I think you meant Trio instead of iAPS.
we still need to start, so months away. Don’t expect something soon.

**Comment by bastiaanv** [comment]
MedtrumKit support is available in feature branch feat/dev-medtrum
https://github.com/nightscout/Trio/tree/feat/dev-medtrum
